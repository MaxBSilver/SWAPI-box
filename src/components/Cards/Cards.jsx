import React from "react";
import Card from "../Card/Card";
import PropTypes from "prop-types";

const Cards = props => {
  return (
    <React.Fragment>
      <h1 className="category-text">
        <div>{props.category.toUpperCase()}</div>
      </h1>
      <section className="cards">
        {props.display.map(item => {
          return (
            <Card
              {...item}
              key={item.Name}
              updateFavorites={props.updateFavorites}
            />
          );
        })}
      </section>
    </React.Fragment>
  );
};

export default Cards;

Cards.propTypes = {
  category: PropTypes.string,
  display: PropTypes.arrayOf(PropTypes.object),
  updateFavorites: PropTypes.func
};
