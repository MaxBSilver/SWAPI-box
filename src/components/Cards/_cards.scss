.cards {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  justify-items: center;
  grid-gap: 2em;
  padding: 2em 2em;
}

main h1 {
  text-align: center;
}

.category-text {
  color: rgb(246, 214, 50);
  text-shadow: -3px -3px 0 #000, 3px -3px 0 #000, -3px 3px 0 #000,
    3px 3px 0 #000;
  font-size: 2.5em;
  letter-spacing: 0.15em;

}

h1.category-text div {
  padding: 0.5em;
  border-bottom: 0.05em solid black;
  background-color: rgb(15, 15, 15);
}

@media only screen and (max-width: 1180px) {
  .cards {
    grid-template-columns: repeat(3, 1fr);
  }
}

@media only screen and (max-width: 915px) {
  .cards {
    grid-template-columns: repeat(2, 1fr);
  }
}
@media only screen and (max-width: 600px) {
  .cards {
    grid-template-columns: repeat(1, 1fr);
  }
}
