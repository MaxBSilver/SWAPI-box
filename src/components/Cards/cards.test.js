import React from 'react';
import Cards from './Cards';
import { shallow } from 'enzyme';
describe('Cards', () => {
    let wrapper;
    let mockProps = {category : 'vehicle', display: [{name: 'Luke'}, {name: 'Obi'}, {name: 'steve'}]};
    beforeEach(() => {
      wrapper = shallow(
        <Cards {...mockProps}
        />
      );
    });
    it('should match the snapshot with all data passed in', () => {
        expect(wrapper).toMatchSnapshot();
      });
  });
