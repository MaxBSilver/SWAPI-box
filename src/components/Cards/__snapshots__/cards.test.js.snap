// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should match the snapshot with all data passed in 1`] = `ShallowWrapper {}`;
